src/Modules/OrchardCore.Commerce/Drivers/ProductPartDisplayDriver.cs

```diff
@@ -48,22 +52,74 @@ public class ProductPartDisplayDriver : ContentPartDisplayDriver<ProductPart>
 
         part.Sku = part.Sku.ToUpperInvariant();
 
+        if (part.ContentItem.As<InventoryPart>() is { } inventoryPart)
+        {
+            part.CanBeBought.Clear();
+
+            var correctInventory = inventoryPart.Inventory
```

Comment:
This invertory correction could be extraced into a method and use that in multiple places.